src/shared/dll_load_notify.c

```diff
@@ -0,0 +1,149 @@
+                                     module_name,
+                                     sizeof(module_name) / sizeof(wchar_t))) {
+                // Check if the images loaded are signed.
+                if (verify_hash_and_pe_signature(module_name) != OS_SUCCESS) {
+#if IMAGE_TRUST_CHECKS == 2
+                    merror_exit("The file '%S' is not signed or its signature is invalid.", module_name);
+#else
+                    mwarn("The file '%S' is not signed or its signature is invalid.", module_name);
+#endif // IMAGE_TRUST_CHECKS == 2
+                } else {
+                    mdebug1("The file '%S' is signed and its signature is valid.", module_name);
+                }
+            }
+        }
+    } else {
+#if IMAGE_TRUST_CHECKS == 2
+        merror_exit("The mechanism of signature validation for loaded modules at startup failed because the modules"
+                    " of the process couldn't be enumerated. Error: %lu", GetLastError());
+#else
+        mwarn("The mechanism of signature validation for loaded modules at startup failed because the modules of the"
+              " process couldn't be enumerated. Error: %lu", GetLastError());
+#endif // IMAGE_TRUST_CHECKS == 2
+
+    }
+#endif // IMAGE_TRUST_CHECKS != 0
+}
+
+/**
+ * @brief Callback function for DLL load notifications
+ * @param reason Reason for the notification.
+ * @param notification_data Data for the notification.
+ * @param context Context for the notification.
+ */
+void CALLBACK dll_notification(ULONG reason,
+                               PLDR_DLL_NOTIFICATION_DATA notification_data,
+                               __attribute__((unused)) PVOID context)
+{
+    //Check for the reason
+    switch(reason)
+    {
+    case LDR_DLL_NOTIFICATION_REASON_LOADED:
+#if IMAGE_TRUST_CHECKS != 0
+        if (verify_hash_and_pe_signature(notification_data->loaded.full_dll_name->Buffer) != OS_SUCCESS) {
+#if IMAGE_TRUST_CHECKS == 2
+            merror_exit("The file '%S' is not signed or its signature is invalid.",
+                        notification_data->loaded.full_dll_name->Buffer);
+#else
+            mwarn("The file '%S' is not signed or its signature is invalid.",
+                  notification_data->loaded.full_dll_name->Buffer);
+#endif // IMAGE_TRUST_CHECKS == 2
+        } else {
+            mdebug1("The file '%S' is signed and its signature is valid.",
+                    notification_data->loaded.full_dll_name->Buffer);
+        }
+#endif // IMAGE_TRUST_CHECKS != 0
+        break;
+    case LDR_DLL_NOTIFICATION_REASON_UNLOADED:
+        mdebug1("Unloaded: '%S'", notification_data->unloaded.full_dll_name->Buffer);
+        break;
+    }
+}
+
+/**
+ * @brief Register for DLL load notifications and verify all the DLLs that are already loaded
+ */
+void enable_dll_verification()
+{
+#if IMAGE_TRUST_CHECKS != 0
+    // Check if all loaded DLLs are signed.
+    if (ERROR_SUCCESS != check_ca_available("DigiCert")) {
```

Comment:
Design question: this will be ALWAYS our CA? how about an
```
#ifndef WAZUH_CA
#define WAZUH_CA "DigiCert"
#endif
```
And a Makefile option that allow to set it

Reply:
In fact: our documentation allows users to sign their own Windows packages https://documentation.wazuh.com/current/deployment-options/wazuh-from-sources/wazuh-agent/index.html. Given the case that the user want to enforce DLL trust mechanism and sign with their custom signature, current mechanism will fail